packages/validator/src/services/syncingStatusTracker.ts

```diff
@@ -0,0 +1,60 @@
+import {ApiClient, routes} from "@lodestar/api";
+import {Logger} from "@lodestar/utils";
+import {Slot} from "@lodestar/types";
+import {IClock} from "../util/clock.js";
+
+export type SyncingStatus = routes.node.SyncingStatus;
+
+type RunOnSyncedFn = (slot: Slot, signal: AbortSignal) => Promise<void>;
+
+/**
+ * Track the syncing status of connected beacon node
+ */
+export class SyncingStatusTracker {
+  private prevSyncingStatus: SyncingStatus | null = null;
+
+  private readonly fns: RunOnSyncedFn[] = [];
+
+  constructor(
+    private readonly logger: Logger,
+    private readonly api: ApiClient,
+    private readonly clock: IClock
+  ) {
+    this.clock.runEverySlot(this.checkSyncingStatus);
+  }
+
+  /** Run function when node status changes from syncing to synced */
+  runOnSynced(fn: RunOnSyncedFn): void {
+    this.fns.push(fn);
+  }
+
+  private checkSyncingStatus = async (slot: Slot, signal: AbortSignal): Promise<void> => {
+    try {
+      const syncingStatus = (await this.api.node.getSyncingStatus()).value();
+
+      if (this.prevSyncingStatus === null) {
+        // Initial check, do not run any handlers
+        this.prevSyncingStatus = syncingStatus;
```

Comment:
looks like this can be pulled out of the conditional

Reply:
need to review this once more, right now if node goes offline it's not correctly setting it as well, might wanna set `prevSyncingStatus` to `null` explicitly if last call failed instead of keeping stale status